pandas/tests/scalar/timestamp/methods/test_replace.py

```diff
@@ -189,3 +189,13 @@ def test_replace_preserves_fold(self, fold):
         ts_replaced = ts.replace(second=1)
 
         assert ts_replaced.fold == fold
+
+    def test_replace_unit(self):
+        # GH#57749
+        ts = Timestamp("2023-07-15 23:08:12")
+        ts1 = Timestamp("2023-07-15 23:08:12.134567")
+        ts2 = Timestamp("2023-07-15 23:08:12.134567123")
+        ts = ts.replace(microsecond=ts1.microsecond)
+        assert ts == ts1
+        ts = ts.replace(nanosecond=ts2.nanosecond)
+        assert ts == ts2
```

Comment:
can you explicitly check the result's unit

Reply:
Hey, I'll make the explicit check and keep the assertions then.